static/js/public/store/filters.js

```diff
@@ -210,17 +200,114 @@ class Filters {
       e.preventDefault();
       this.removeFilter("sort");
       this.addFilter("sort", el.value);
+      this.syncSortUI("sortMobile", el.value);
+
+      this.cleanFilters();
+      this.updateHistory();
+    });
+  }
+
+  initMobileSortEvents(el) {
+    el.addEventListener("change", (e) => {
+      e.preventDefault();
+      this.removeFilter("sort");
+      this.addFilter("sort", e.target.value);
+      this.syncSortUI("sort", e.target.value);
 
       this.cleanFilters();
       this.updateHistory();
+
+      // hide the drawer once clicked
+      el.classList.remove("is-active");
+    });
+  }
+
+  initFilterEvents(el) {
+    const filters = el.querySelectorAll("input");
+    const resetButton = el.querySelector("[data-js='filter-reset']");
+    const submitButton = el.querySelector("[data-js='filter-submit']");
+
+    if (filters) {
+      filters.forEach((filterEl) => {
+        filterEl.addEventListener("click", () => {
```

Comment:
Adding event listeners in a loop like this to individual elements can add overhead, that's why the original initFilterEvents used [event delegation](https://javascript.info/event-delegation). Could this be changed back please?